packages/react-aria-components/docs/Button.mdx

```diff
@@ -159,6 +159,136 @@ A `Button` can be disabled using the `isDisabled` prop.
 
 </details>
 
+## Pending
+
+A `Button` can be put into the pending state using the `isPending` prop.
+A [ProgressBar](ProgressBar.html) component is required to show the pending state correctly.
+Make sure to internationalize the label you pass to the [ProgressBar](ProgressBar.html) component.
+
+```tsx example
+import {useRef, useState} from 'react';
+import {ProgressBar, Text} from 'react-aria-components';
+
+function PendingButton(props) {
+  let [isPending, setPending] = useState(false);
+
+  let timeout = useRef<ReturnType<typeof setTimeout> | undefined>(undefined);
+  let handlePress = (e) => {
+    setPending(true);
+    timeout.current = setTimeout(() => {
+      setPending(false);
+      timeout.current = undefined;
+    }, 5000);
+  };
+
+  return (
+    <Button
+      {...props}
+      isPending={isPending}
+      onPress={handlePress}>
+        {({isPending}) => (
+          <>
+            {!isPending && <Text>Save</Text>}
+            {isPending && (<ProgressBar
+              aria-label="Saving..."
+              isIndeterminate>
+              <svg width="24" height="24" viewBox="0 0 24 24" style={{display: 'block'}}>
+                <path fill="currentColor" d="M12,1A11,11,0,1,0,23,12,11,11,0,0,0,12,1Zm0,19a8,8,0,1,1,8-8A8,8,0,0,1,12,20Z" opacity=".25" />
+                <path fill="currentColor" d="M10.14,1.16a11,11,0,0,0-9,8.92A1.59,1.59,0,0,0,2.46,12,1.52,1.52,0,0,0,4.11,10.7a8,8,0,0,1,6.66-6.61A1.42,1.42,0,0,0,12,2.69h0A1.57,1.57,0,0,0,10.14,1.16Z">
+                  <animateTransform attributeName="transform" type="rotate" dur="0.75s" values="0 12 12;360 12 12" repeatCount="indefinite"/>
+                </path>
+              </svg>
+            </ProgressBar>)}
+          </>
+        )}
+    </Button>
+  );
+}
+<PendingButton />
+```
+
+**Note:**
+The `ProgressBar` must be in the accessibility tree when the button becomes pending.
+This can be achieved by using opacity instead of visibility or display to hide the spinner if you'd like it wait a moment before showing.
+The `Text` can remain in the button when the button is pending, in which case you should decide if it stays in the accessiblity
+tree or if you should hide it in favor of the label on the `ProgressBar`.
```

Comment:
I think we may want to explain this more. Rather than starting with the caveats, maybe start with what we're trying to achieve.

**Note:** The `ProgressBar` must be in the accessibility tree as soon as the button becomes pending, even if it is not visible. For example, if you'd like to delay showing a spinner until a minimum amount of time passes, you could use `opacity: 0` to hide it so it is still available to screen readers. Do not use `visibility: hidden` or `display: none` since these remove the element from the accessibility tree.

Additionally, you may choose to keep the button's contents in the DOM while the button is pending, e.g. to preserve the button's layout. If you hide the contents with `visibility: hidden`, the accessibility label for the button will only include the `ProgressBar`, so it should have a descriptive `aria-label` (e.g. "Saving"). You can also choose to keep the button's contents in the accessibility tree by using `opacity: 0`, in which case the `ProgressBar`'s label will be combined with the contents (e.g. "Save, pending").

(maybe a little long, feel free to edit)